### DISCUSSION TITLE
Help with i2c screen that only show black squares

### DISCUSSION THREAD

**Discussion by EphraimElgrabli** [probonopd/MiniDexed#706] [Q&A] [answered]
I recently bought a new display for my project to replace the ST7789 screen I had before. I specifically purchased this screen, which seems to be supported: [AliExpress Link](https://www.aliexpress.com/item/1005006018109299.html?spm=a2g0o.order_list.order_list_main.29.5f0e586aCChPFI)
However, I'm experiencing an issue. When I plug the power cable into the Raspberry Pi Zero W 2, the screen boots up but only shows black squares. I've checked if the Pi is faulty by running RetroPie with an HDMI signal, tested several SD cards formatted to FAT32, and checked my soldered pins multiple times. I've also configured the INI file with all possible communication addresses (0x20 - 0x27), but still have had no luck.
Can you help me troubleshoot this problem? Thank you in advance.

**Comment by craigyjp** [comment]
What's your config file look like, have you configured it for i2c, usually white squares is a contrast issue. 

Also is it connected to +5v or +3.3v

**Reply by EphraimElgrabli** [reply]
I've taken care of the contrast by adjusting the potentiometer to get a nice contrast.
Which config file are you referring to? Is it the minidexed.ini?
connected to 5v

**Reply by craigyjp** [reply]
Yeah, the MiniDexed.ini file, it would be useful to see that. Also do you have an Arduino or something available? You could quickly connect the display to the Arduino and run an i2c finder to determine the address and maybe an bit of example code that talks to the LCD on i2c after you have correctly identified the address on the i2c bus.

**Reply by EphraimElgrabli** [reply]
HD44780 LCD
LCDEnabled=1
LCDPinEnable=17
LCDPinRegisterSelect=4
LCDPinReadWrite=0
LCDPinData4=22
LCDPinData5=23
LCDPinData6=24
LCDPinData7=25
LCDI2CAddress=0x27

I tried all the possible adresses (0x20 - 0x27). I have an esp32, is it useful?

**Reply by craigyjp** [reply]
That doesn't look right for an i2c connection, it does for an standard parallel interface, but i2c is different.

**Reply by EphraimElgrabli** [reply]
what should I do?

**Reply by craigyjp** [reply]
First connect it to your esp32 on the i2c pins, download an i2c finder and see what is found at what address, that will definitely prove the existence of an i2c device at a know address, find a library for and LCD with i2c interface, look at an example and run it with the correct address, things should appear on the LCD screen if the address matches the code.

I would look for an i2c config for your setup, are the sda and scl pins on the correct pins for the pi

**Reply by craigyjp** [reply]
On the pi you really only need the i2c line in the config for the address, the data lines for LCD are not required. 

The sda and scl pins should connect to your i2c interface along with +5v and ground.

**Reply by EphraimElgrabli** [reply]
Ok I will try this approach, thank you so much!

**Reply by craigyjp** [reply]
Read this section regarding LCD and back packs there are two address ranges...

https://github.com/probonopd/MiniDexed/wiki/Hardware

**Reply by diyelectromusic** [reply, marked as answer]
> That doesn't look right for an i2c connection, it does for an standard parallel interface, but i2c is different.

If LCDI2CAddress is defined then the other entries are ignored.  LCDEnabled does still need to be set to 1 though.

And yes, the two ranges for these devices.  If it uses the PCF8574 chip then the range is 0x20-0x27 (0x27 usually the default) or 0x38-0x3F (0x3F as default) for devices with a PCF8574A.

Kevin

**Reply by craigyjp** [reply]
According to the AliExpress link it's a T variant of the chip, I don't know what that means to addressing, if he connects it to the esp32 and scans the i2c bus it should show it's address.
